<?xml version="1.0" encoding="UTF-8"?>
<launch>

	<!-- <arg name="launch_prefix" default="gdb -ex run -args" /> -->
	<arg name="launch_prefix" default="" />

	<arg name="icp_config_name" default="icp.yaml" />
	<arg name="icp_config_path"
		default="$(find icp_localization)/config/$(arg icp_config_name)" />

	<arg name="input_filters_config_name" default="input_filters.yaml" />
	<arg name="input_filters_config_path"
		default="$(find icp_localization)/config/$(arg input_filters_config_name)" />

	 <arg name="pcd_filename" default="compslam_map.pcd" /> 
	 <arg name="pcd_filepath" 
		default="$(find icp_localization)/data/$(arg pcd_filename)" /> 

	<arg name="launch_rviz" default="false" />


	<!-- Launch icp node -->
	<node name="icp_node" pkg="icp_localization" type="localizer_node"
		output="screen" launch-prefix="$(arg launch_prefix)">
		<param name="pcd_filename" value="$(arg pcd_filepath)" />
		<rosparam command="load"
			file="$(find icp_localization)/config/param.yaml" />
		<param name="icp_config_path" value="$(arg icp_config_path)" />
		<param name="input_filters_config_path"
			value="$(arg input_filters_config_path)" />
	</node>


  <include file="$(find icp_localization)/launch/icp_vis.launch" if="$(arg launch_rviz)">
  </include>


</launch>